transforms/geocable.py

```diff
@@ -1213,11 +1213,44 @@ async def compile_rope(
             if node.config.coll_side_count >= 3:
                 has_coll = True
 
-        model_name, (light_origin, coll_data, seg_props, vac_points) = await compiler.get_model(
-            (frozenset(local_nodes), frozenset(connections)),
-            build_rope,
-            (center, ctx.pack.fsys),
-        )
+        class ModelContainer:
+            def __init__(self,mn,lo,cd,sp):
+                self.model_name = mn
+                self.light_origin = lo
+                self.coll_data = cd
+                self.seg_props = sp
+
+        modellist : List[ModelContainer] = []
+        # Separate glass config
+        # Call get_model twice to get models for frame and glass combined
+
+        # This code need refactorization
+        if node.config.vac_separate_glass == VactubeGenType.SEPARATE:
+
+            # generate glass only
+            model_name, (light_origin, coll_data, seg_props, _) = await compiler.get_model(
+                (frozenset(local_nodes), frozenset(connections), VactubeGenPartType.GLASS),
+                build_rope,
+                (center, ctx.pack.fsys),
+            )
+            #hmmmmm
+            modellist.append(ModelContainer(model_name, light_origin, coll_data, seg_props))
+            
+            # generate frame only
+            # frame generation code also generates the vac_points
+            model_name, (light_origin, coll_data, seg_props, vac_points) = await compiler.get_model(
+                (frozenset(local_nodes), frozenset(connections), VactubeGenPartType.FRAME),
+                build_rope,
+                (center, ctx.pack.fsys),
+            )
+            modellist.append(ModelContainer(model_name, light_origin, coll_data, seg_props))
+        else: 
+            model_name, (light_origin, coll_data, seg_props, vac_points) = await compiler.get_model(
+                (frozenset(local_nodes), frozenset(connections), VactubeGenPartType.ALL),
+                build_rope,
+                (center, ctx.pack.fsys),
+            )
+            modellist.append(ModelContainer(model_name, light_origin, coll_data, seg_props))
```

Comment:
Actually, maybe this should be more like:
```suggestion
        is_sep = node.config.vac_separate_glass == VactubeGenType.SEPARATE

        # Generate the whole model, or frame only if separating. 
        model_name, (light_origin, coll_data, seg_props, vac_points) = await compiler.get_model(
            (frozenset(local_nodes), frozenset(connections), VactubeGenPartType.FRAME if is_sep else VactubeGenPartType.ALL),
            build_rope,
            (center, ctx.pack.fsys),
        )
        modellist.append(ModelContainer(model_name, light_origin, coll_data, seg_props))

        if is_sep:
            # Generate the glass only
            model_name_gls, (light_origin, coll_data, seg_props, _) = await compiler.get_model(
                (frozenset(local_nodes), frozenset(connections), VactubeGenPartType.GLASS),
                build_rope,
                (center, ctx.pack.fsys),
            )
            modellist.append(ModelContainer(model_name, light_origin, coll_data, seg_props))
```